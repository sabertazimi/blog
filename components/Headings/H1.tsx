import cx from 'classnames'
import type { ReactNode } from 'react'
import Heading from './Heading'
import styles from './Headings.module.css'

interface Props {
  className?: string
  children?: ReactNode
}

function H1({ className, ...props }: Props): JSX.Element {
  return <Heading {...props} level={1} className={cx(className, styles.h1)} />
}

export default H1
