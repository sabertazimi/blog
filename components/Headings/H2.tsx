import { cx } from '@components/utils'
import type { ReactNode } from 'react'
import Heading from './Heading'
import styles from './Headings.module.css'

interface Props {
  className?: string
  children?: ReactNode
}

const H2 = ({ className, ...props }: Props): JSX.Element => (
  <Heading {...props} level={2} className={cx(className, styles.h2)} />
)

export default H2
