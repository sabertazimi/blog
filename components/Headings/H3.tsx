import { cx } from '@components/utils'
import type { ReactNode } from 'react'
import Heading from './Heading'
import styles from './Headings.module.css'

interface Props {
  className?: string
  children?: ReactNode
}

function H3({ className, ...props }: Props): JSX.Element {
  return <Heading {...props} level={3} className={cx(className, styles.h3)} />
}

export default H3
