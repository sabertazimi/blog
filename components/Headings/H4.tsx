import { cx } from '@components/utils'
import type { ReactNode } from 'react'
import Heading from './Heading'
import styles from './Headings.module.css'

interface Props {
  className?: string
  children?: ReactNode
}

function H4({ className, ...props }: Props): JSX.Element {
  return <Heading {...props} level={4} className={cx(className, styles.h4)} />
}

export default H4
