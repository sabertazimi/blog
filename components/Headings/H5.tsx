import cx from 'classnames'
import type { ReactNode } from 'react'
import Heading from './Heading'
import styles from './Headings.module.css'

interface Props {
  className?: string
  children?: ReactNode
}

function H5({ className, ...props }: Props): JSX.Element {
  return <Heading {...props} level={5} className={cx(className, styles.h5)} />
}

export default H5
