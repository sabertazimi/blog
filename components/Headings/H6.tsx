import { cx } from '@components/utils'
import type { ReactNode } from 'react'
import Heading from './Heading'
import styles from './Headings.module.css'

interface Props {
  className?: string
  children?: ReactNode
}

const H6 = ({ className, ...props }: Props): JSX.Element => (
  <Heading {...props} level={5} className={cx(className, styles.h6)} />
)

export default H6
