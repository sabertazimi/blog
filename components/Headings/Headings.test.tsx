import { render } from '@utils'
import H1 from './H1'
import H2 from './H2'
import H3 from './H3'
import H4 from './H4'
import H5 from './H5'
import H6 from './H6'

describe('Headings', () => {
  const Headings = {
    h1: H1,
    h2: H2,
    h3: H3,
    h4: H4,
    h5: H5,
    h6: H6,
  }

  test.each(Object.values(Headings))(
    'should render correctly (snapshot)',
    Heading => {
      const { container } = render(<Heading>Heading</Heading>)

      expect(container).toMatchSnapshot()
    }
  )
})
