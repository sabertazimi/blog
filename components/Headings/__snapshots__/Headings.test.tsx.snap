// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`headings should render correctly (snapshot) 1`] = `
<div>
  <h1
    class="ant-typography mdx-heading dark:text-light heading h1"
  >
    Heading
  </h1>
</div>
`;

exports[`headings should render correctly (snapshot) 2`] = `
<div>
  <h2
    class="ant-typography mdx-heading dark:text-light heading h2"
  >
    Heading
  </h2>
</div>
`;

exports[`headings should render correctly (snapshot) 3`] = `
<div>
  <h3
    class="ant-typography mdx-heading dark:text-light heading h3"
  >
    Heading
  </h3>
</div>
`;

exports[`headings should render correctly (snapshot) 4`] = `
<div>
  <h4
    class="ant-typography mdx-heading dark:text-light heading h4"
  >
    Heading
  </h4>
</div>
`;

exports[`headings should render correctly (snapshot) 5`] = `
<div>
  <h5
    class="ant-typography mdx-heading dark:text-light heading h5"
  >
    Heading
  </h5>
</div>
`;

exports[`headings should render correctly (snapshot) 6`] = `
<div>
  <h5
    class="ant-typography mdx-heading dark:text-light heading h6"
  >
    Heading
  </h5>
</div>
`;
